### PR TITLE
add jsoncpp package

### PR DESCRIPTION
**Pull request by Ubiquite**
I'm adding jsoncpp library : https://github.com/open-source-parsers/jsoncpp

### CODE REVIEW

File: cmake/projects/jsoncpp/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# cmake/projects/jsoncpp/hunter.cmake
+
+# !!! DO NOT PLACE HEADER GUARDS HERE !!!
+
+# Load used modules
+include(hunter_add_version)
+include(hunter_download)
+include(hunter_pick_scheme)
```
Comment (ruslo) [inline cmake/projects/jsoncpp/hunter.cmake]: We need `include(hunter_cmake_args)`. You can run test by this script:
* https://github.com/ruslo/hunter/wiki/dev.hunter.root#jenkinspy

**Comment by ruslo** [comment]
https://github.com/ruslo/hunter/issues/256

File: cmake/projects/jsoncpp/hunter.cmake (hunk 1 above)
Comment (Ubiquite) [inline cmake/projects/jsoncpp/hunter.cmake]: added
also the test run successfully

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1333
* https://travis-ci.org/ingenue/hunter/builds/199139884

**Comment by ruslo** [comment]
> Testing

Some tests failed. @Ubiquite please take a look - we need to make everything "green" either by fixing issues (updating package) or by excluding broken toolchains. In second case please send pull request to branch https://github.com/ingenue/hunter/tree/pkg.jsoncpp Also see https://github.com/ingenue/hunter/blob/pkg.template/README.rst

**Comment by Ubiquite** [comment]
at least for vs-9-2008 we have to set default version to 0.7.0 or disable jsoncpp. 1.7.7 requires some C++11. But how do you do this usually ?

**Comment by ruslo** [comment]
> at least for vs-9-2008 we have to set default version to 0.7.0 or disable jsoncpp

You can use MSVC_VERSION.

> 1.7.7 requires some C++11. But how do you do this usually ?

By toolchain. `default` doesn't set any flags, meaning `c++98` will be used. Other toolchains use `c++11` but some `c++11` library's features may be missing for particular toolchains. 

If the problem is not about package, just exclude matrix entry.

**Comment by ruslo** [comment]
Testing:
* https://travis-ci.org/ingenue/hunter/builds/199581411
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1337

File: cmake/configs/default.cmake [hunk 2]
```diff
@@ -60,6 +60,12 @@ hunter_config(Igloo VERSION 1.1.1-hunter)
 hunter_config(IntSizeof VERSION 1.0.0)
 hunter_config(Jpeg VERSION 9b-p1)
 hunter_config(JsonSpirit VERSION 0.0.4-hunter)
+if(MSVC_VERSION LESS 1600)
+    # for VS10 - version without support C++11
+  hunter_config(jsoncpp VERSION 0.7.0)
```
Comment (ruslo) [inline cmake/configs/default.cmake]: Still doesn't work:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1337/job/0xrax3cr46clgb9p

Comment (Ubiquite) [inline cmake/configs/default.cmake]: Should be all green now :
the name of the imported library generated by find_package(jsoncpp... is not the same with 0.7.0 and 1.7.7. So I added a if(.. statement in the jsoncpp test

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1338
* https://travis-ci.org/ingenue/hunter/builds/199786006

**Comment by ruslo** [comment]
Released https://github.com/ruslo/hunter/releases/tag/v0.17.19

**Pull request merged**